### ISSUE TITLE
Native base components make app slower by using FowardRef

### ISSUE THREAD

**Issue by Wei102193** [GeekyAnts/NativeBase#4681]
### Description

My app is mostly written with native base, it is quick and convenient. But somehow it is really slow on android. I used React Native Debugger Profiler and found out that those native base components are all using FowardRef, which seems to make app render slower. I actually rewrote one of the screen with purely react native and the render duration decreased from 180ms to 80ms. Anyone notices this issue?

### CodeSandbox/Snack link

no code sandbox link required

### NativeBase Version

3.3.1

### Platform

- [X] Android
- [ ] CRA
- [X] Expo
- [X] iOS
- [ ] Next

### Additional Information

![Screen Shot 2022-02-28 at 5 44 15 PM](https://user-images.githubusercontent.com/27971322/156070810-c90597a2-1a9a-46c1-b91a-81679a4c9be8.png)

**Comment by surajahmed** [comment]
@Wei102193 Thanks for reporting! We'll look into this.

**Comment by ajaygupta1988** [comment]
I am having the same issue. I rewrote my entire react-native app with native-base because of its simplicity but when we did our final preproduction testing, the app is running very slow on android. For ios, it seems to be working fine maybe it is slower than pure react-native but the difference is not noticeable on ios.

**Comment by pierroo** [comment]
> @Wei102193 Thanks for reporting! We'll look into this.

Is there any way to track the progress about performance?
Right now it is indeed crazy slow on android in production; I would love to see some follow up on this issue.

**Comment by Wei102193** [comment]
> > @Wei102193 Thanks for reporting! We'll look into this.
> 
> Is there any way to track the progress about performance? Right now it is indeed crazy slow on android in production; I would love to see some follow up on this issue.

Yes, you can use Profiler from React Native Debugger to check the render duration. As I said I rewrote one of my screen with only React Native, the render duration decreased from 180ms to 80ms. It was not even a complex screen. I think the more Native Base components you use, the slower you app will be, especially on Android.
https://docs.expo.dev/workflow/debugging/
https://www.youtube.com/watch?v=00RoZflFE34

**Comment by pierroo** [comment]
I'm sorry I should have clarified, by tracking I meant traking the progress of the native-base team on this issue.
@surajahmed 

Your post was already very clear @Wei102193 and I confirmed I faced the same issue.

**Comment by ajaygupta1988** [comment]
Hi @surajahmed 

Can you please update us on this issue? or at least have a little more clarity regarding its validity.

The application I am working on is near complete and we have to push it to production but if this issue is valid and we have to stop and recode everything again.

I hope you understand my problem. Thanks

**Comment by pierroo** [comment]
Okay here is my update after rewriting most components in pure react native:
performance is day and night. It feels SO MUCH SMOOTHER without native-base, it's unbelievable.

HOWEVER, I am still using nativebase for a few specific components, for dark mode and a few extra stuff.
But, as soon as I have a list of many items/components, then right away I switch said components to react-native, and that itself should save you a lot of performance without necessarily writing your entire code.

Basically, keep wrappers etc with native vase, but when you are coding a list of items, make sure said list is react native.
It's 20% effort for 80% result for the time being, while waiting for native base to improve performances.

**Comment by ajaygupta1988** [comment]
@pierroo Thank you for sharing your solution. Seeing all the comments above, I assume this problem is actually valid.

I think fixing this issue should be on the top priority list of the native-base team @surajahmed. I personally loved the library and the approach but it's sad that we can't use it in production as-is for now.

I hope this issue will be resolved soon.

**Comment by frossi85** [comment]
While the issue described here is regarding Mobile, I am experiencing a similar issue using native base with NextJS. The biggest issue is observed on inputs that use text like Input and TextArea. Writing fast on the input does not provide instant feedback but takes a while between typing fast and seeing what was typed reflected in those inputs. Also, there are some slow animations. So I guess the issues are in the core and not the platform-specific implementation.

Does the team have a deadline? I am asking because started to implement something that needs to go to prod in 2 months but if the issue remains I will need to replace Nativebase.

**Comment by rayan1810** [comment]
Closing this issue as it is actually the same issue as https://github.com/GeekyAnts/NativeBase/issues/4302, Please follow up on it there, Thanks.
